[12.x] add type declarations for Console Events

browner12 opened the pull request:
the added types match the existing constructor docblocks.

not sure what the current sentiment on stricter typing is from the core team, but thought I'd throw this out there. these events are a good place to start as their construction should basically only be internal.

I checked all the calling code, and this may have already uncovered a bug. the `CommandStarting` claims to accept a string `$command` parameter, but the calling code possibly sends `null`.  two ways to fix this would be to accept `string|null`, or to have the calling code null coalesce and empty string (`?? ''`).

if accepted, I'll update all the other events as well.

<!--
Please only send a pull request to branches that are currently supported: https://laravel.com/docs/releases#support-policy 

If you are unsure which branch your pull request should be sent to, please read: https://laravel.com/docs/contributions#which-branch

Pull requests without a descriptive title, thorough description, or tests will be closed.

In addition, please describe the benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.
-->

browner12 committed:
add type declarations for Console Events

the added types match the existing constructor docblocks

browner12 committed:
minor styling

shaedrich commented:
I predict, this will be rejected due to backwards compatibility

taylorotwell commented:
@browner12 under what situation does `CommandStarting` currently receive `null`?

browner12 commented:
@taylorotwell I don't know if it **actually** happens, but in

https://github.com/laravel/framework/blob/11.x/src/Illuminate/Foundation/Console/Kernel.php#L166

the `getCommand()` can return `null`, and the `getName()` can also return `null`. my guess is it never actually happens, but based on their declared return types they are nullable.

taylorotwell commented:
Hmm, ok, I guess we should coalesce in the callers then so the event itself only receives stirngs.

browner12 committed:
handle nullable `getCommand()` and `getName()`

both the `CommandStarting` and `CommandFinished` require a string to be passed as the first argument, so we need to account for nullable scenarios.

browner12 commented:
@taylorotwell code has been updated. I realized the `CommandFinished` also required a string.

Very weird on symfonys part, how could you have a command event without a command?

crynobone commented:
> Very weird on symfonys part, how could you have a command event without a command?

```bash
php artisan --version
```

This could be one example.

browner12 commented:
> > Very weird on symfonys part, how could you have a command event without a command?
> 
> ```shell
> php artisan --version
> ```
> 
> This could be one example.

if that's not a command, then why is it firing off a command event?

I'm sure they had good reason, just seems odd from the outside looking in.
